controller/testing/fake_stats_reporter_test.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/knative/pkg/controller"
+)
+
+var _ controller.StatsReporter = (*FakeStatsReporter)(nil)
```

Comment:
See my other comment, I'd put this right after the type definition.

Reply:
That causes a cyclic dependency between controller and testing packages. controller_test.go make use of fake_stats_reporter. And if I move that line to fake_stats_reporter.go, it will end up importing controller package, which causes the error.

I can alternatively create another folder, or define the interface in a common folder, but that seems too much overhead for just a test utility. Instead, I put it here in the _test.go file and while this won't cause `go build` failure, it will still cause `go test` failure.